### DISCUSSION TITLE
How to do zero-time deployment?

### DISCUSSION THREAD

**Discussion by lezhnev74** [roadrunner-server/roadrunner#1757] [Q&A] [answered]
Hey! I was wondering if there is a tutorial explaining how to setup deployment scripts to achieve 0-time deployment. Like when we want to roll out a new version of our app on a server, we probably want to start a new docker container, pass some secrets, wait until the service is ready and then switch traffic to it. Do you have any docs on that matter? Thanks.

**Comment by rustatian** [comment, marked as answer]
Hey @lezhnev74 👋🏻 
We don't have such a doc, but any documentation about canary or green-blue deployment will work. RR doesn't have specifics which prevent using any modern techniques.

**Reply by lezhnev74** [reply]
yeah, good point. Thank you for a quick response :+1:
